### PR TITLE
Enhance bulk email functionality

### PR DESCRIPTION
**Pull request by benthorner**
https://trello.com/c/wp7WjPN6/693-draft-email-to-send-out-to-checker-subscribers-following-an-announcement

This makes a few changes to the existing bulk email builder:

- It is now batches generation of large quantities of email.
- The email now ends with a standard unsubscribe / manage footer.
- A list URL can be substituted for a "%LISTURL%" placeholder.

Arguably the builder is now doing too much: it should not be
deciding which emails to generate, but simply generating them.
This has lead to the test setup becoming a bit convoluted. In
future work, we should refactor the builder to separate out the
selection of email, like we do for immediate emails [1].

## Before (Link goes to "%LISTURL%" i.e. doesn't work)

Screenshot 2021-01-04 at 14 54 06

## After (link goes to checker results page for subscriber)

Screenshot 2021-01-04 at 14 44 07

[1]: https://github.com/alphagov/email-alert-api/blob/d248a8d63324cb2255f6db36104cce9f3ed0bf2f/app/services/immediate_email_generation_service.rb

### CODE REVIEW

File: app/builders/bulk_subscriber_list_email_builder.rb
Comment (hannako) [inline app/builders/bulk_subscriber_list_email_builder.rb:25]: Feel free to ignore:
I'm not v familiar with email alert api. It took me a while to figure out that calling `.to_h` on the array produced by the scope `.subscription_ids_by_subscriber` would mean that `batch.key` was a subscriber id. and like wise `batch.values` are subscription ids. Perhaps `batch.keys` and `batch.values` could be their own more explicitly named methods?

**Review by hannako — APPROVED** [review]
LGTM

File: app/builders/bulk_subscriber_list_email_builder.rb
Comment (benthorner) [reply app/builders/bulk_subscriber_list_email_builder.rb]: Yes, it's pretty obscure isn't it! Apparently it's [a Rails-ism](https://apidock.com/ruby/v2_5_5/Array/to_h), which I copied from [here](https://github.com/alphagov/email-alert-api/blob/master/app/services/immediate_email_generation_service.rb#L36). I'm hoping to make this clearer in the follow-up PR, but for now I've amended [the fourth commit](https://github.com/alphagov/email-alert-api/pull/1542/commits/c872fa49f54eebb2bb67f6054edd254ca2c27387) so it's hopefully a little clearer. 

How does it look now?

**Comment by hannako** [comment]
Have looked at the commits made after approval, and still looks good 👍

**Pull request merged**
